### PR TITLE
[feat]: add twitter share button

### LINKED ISSUE #13: [FEATURE] Share button for twitter
Create an interactive share to Twitter button.
don't forget to add up the quote with it, so that the quote with author name, can be share to Twitter.

### PR DESCRIPTION
**Pull request by adit-shriyans**
# PULL REQUEST TEMPLATE

**My PR closes #13 **

### 👨‍💻 What did you do ?

-Added a share to twitter(X) button
-Fixed Whatsapp share button

### Note to reviewers

# 📷 Screenshots

image
image

### ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I have read the [contribution guideline](https://github.com/Shariar-Hasan/QuoteVerse/blob/main/Contributing.md) properly and following the criteria.

### CODE REVIEW

**Comment by Shariar-Hasan** [comment]
you should not fixed the whatsapp button.
now resolve the conflict.

File: index.html [hunk 1]
```diff
@@ -128,10 +135,12 @@ <h5 class="quote-author" id="quoteAuthorTag">- Steve Jobs</h5>
...
     })
+
     twtbtn.addEventListener("mouseenter", () => {
-      let quoteBody = document.querySelector(".quote .quote-text").innerHTML;
-      let quoteAuthor = document.querySelector(".quote .quote-author").innerHTML;
+      let quoteBody = document.querySelector(".quote .quote-text").innerText;
+      let quoteAuthor = document.querySelector(".quote .quote-author").innerText;
       let postText = quoteBody + "%0A" + quoteAuthor + "%0A" + "I found it here:" + postUrl;
+      console.log("QUOTE BODY:", quoteBody, "\nQUOTE AUTH:", quoteAuthor, "\nPOST:", postText);
       postText = postText.replace(/ /g, "%20");
```
Comment (Shariar-Hasan) [inline index.html:144]: delete the console log

File: index.html [hunk 2]
```diff
@@ -74,16 +74,22 @@ <h5 class="quote-author" id="quoteAuthorTag">- Steve Jobs</h5>
...
-          <a href="#" class="whatsapp-btn">
-            <i class="fa-brands fa-square-whatsapp fa-2xl" style="color: #00ff33;"></i>
-          </a>
+          <div class="washare">
+            <a href="#" class="whatsapp-btn">
+              <i class="fa-brands fa-square-whatsapp fa-2xl" style="color: #00ff33;"></i>
+            </a>
+          </div>
           <div class="fbshare">
             <i class="fa-brands fa-facebook fa-2xl" style="color: #0084ff;"></i>
```
Comment (Shariar-Hasan) [inline index.html:83]: no need to fix whatapp share button.
dont do anything unless you are assign to do.
you should have read the contribution rules.

File: index.html [hunk 3]
```diff
@@ -111,15 +117,16 @@ <h5 class="quote-author" id="quoteAuthorTag">- Steve Jobs</h5>
       })
     })
     whatsappBtn.addEventListener("click", () => {
-      let quoteBody = document.querySelector(".quote .quote-text").innerHTML;
-      let quoteAuthor = document.querySelector(".quote .quote-author").innerHTML;
+      let quoteBody = document.querySelector(".quote .quote-text").innerText;
+      let quoteAuthor = document.querySelector(".quote .quote-author").innerText;
       let postTitle = quoteBody + quoteAuthor + " " + "\nLink:";
       console.log(quoteBody, quoteAuthor, postUrl);
```
Comment (Shariar-Hasan) [inline index.html:123]: resolve the conflict

File: index.html [hunk 4]
```diff
@@ -125,8 +135,22 @@ <h5 class="quote-author" id="quoteAuthorTag">- Steve Jobs</h5>
...
+      let quoteBody = document.querySelector(".quote .quote-text").innerText;
+      let quoteAuthor = document.querySelector(".quote .quote-author").innerText;
+      let postText = quoteBody + "%0A" + quoteAuthor + "%0A" + "I found it here:" + postUrl;
+      console.log("QUOTE BODY:", quoteBody, "\nQUOTE AUTH:", quoteAuthor, "\nPOST:", postText);
+      postText = postText.replace(/ /g, "%20");
+      if (!twtcount) {
+        console.log(twtbtn.innerHTML);
+        twtbtn.innerHTML = `<div class="twt-share-button" data-href="https://quote-verse.netlify.app/" data-layout="" data-size=""><a target="_blank" class="twitter-share-button" href=${`https://twitter.com/intent/tweet?text=${postText}`} data-size="large"><i class="fa-brands fa-twitter fa-2xl" style="color: #5f99fc;"></i></a></div>`;
+        console.log("after", twtbtn)
+      }
```
Comment (Shariar-Hasan) [inline index.html:149]: remove the console log

**Comment by Shariar-Hasan** [comment]
Never ever do anything more than whatever you assigned to do.
not in any repository.
i understand you wanted to make the UI more better.
but everything has a process,
someone else has fixed the sharebutton for whatsapp,
according to contribution rules, your pull request should be close.
if you do  not know how to resolve the issue,
just copy your code somewhere, whatever you have changed.
you just sync the fork repo, and git pull in vscode.
then paste your needed code in where you want.

**Comment by adit-shriyans** [comment]
I just saw the pr where someone else has fixed the whatsapp share button, i'll get all the things fixed

**Comment by Shariar-Hasan** [comment]
> I saw the pr fixing whatsapp share after my pr, which the other person made just before my pr, i'll get all the things fixed

yes, but he raised an issue about the whatsapp share problem,
then solved it.

File: index.html [hunk 5]
```diff
@@ -80,6 +80,11 @@ <h5 class="quote-author" id="quoteAuthorTag">- Steve Jobs</h5>
           <div class="fbshare">
             <i class="fa-brands fa-facebook fa-2xl" style="color: #0084ff;"></i>
           </div>
+          <div class="twtshare">
+            <a class="twitter-share-button" href="https://twitter.com/intent/tweet?text=Hello%20world">
+              <i class="fa-brands fa-twitter fa-2xl" style="color: #5f99fc;"></i>
```
Comment (iHassan) [inline index.html:85]: Just a suggestion. Update twitter logo with the X logo

```suggestion
              <i class="fa-brands fa-x-twitter fa-2xl" style="color: #5f99fc;"></i>
```

**Review by Shariar-Hasan — APPROVED** [review]
impressive work 💥💥
seems all good🆗
merged✅

**Pull request merged**